crates/sui-adapter-transactional-tests/tests/upgrade/multi_version.move

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//# init --addresses Test=0x0 --accounts A
+
+//# publish --upgradeable --sender A
+module Test_V1::M1 {
+    use sui::tx_context::TxContext;
+    fun init(_ctx: &mut TxContext) { }
+}
+
+//# upgrade --package Test_V1 --new-package-name Test_V2 --upgrade-capability 1,1 --sender A
```

Comment:
> That is having multiple versions of the same package name around?

The main motivation (for me) is to support programmable transactions that call into multiple versions of a package -- that's an important test of how link context works.

Reply:
Also--we don't need `--new-package-name` and can derive it from the module in the test (just updated PR removing the need for `--new-package-name`)

Reply:
> The main motivation (for me) is to support programmable transactions that call into multiple versions of a package -- that's an important test of how link context works.

That was my guess, hence my suggestion. I don't think we need `--package` or `--new-package-name`

Reply:
I do think there's an element of magic involved in automatically generating a different name for an existing object because of an update -- I don't know how I would figure out what was happening by looking at an existing test there, but if it is really tedious to work with, I don't mind adding that in place of the --package flag.